src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxePsiFieldImpl.java

```diff
@@ -213,4 +208,16 @@ public boolean hasModifierProperty(@HaxePsiModifier.ModifierConstant @NonNls @No
     return this.getModifierList().hasModifierProperty(name);
   }
 
+  @NotNull
+  @Override
+  public SearchScope getUseScope() {
+    final PsiElement localVar = UsefulPsiTreeUtil.getParentOfType(this, HaxeLocalVarDeclaration.class);
+    if(localVar != null) {
+      final PsiElement outerBlock = UsefulPsiTreeUtil.getParentOfType(localVar, HaxeBlockStatement.class);
```

Comment:
And then, after making these arguments, I go look at your unit test more closely and it's working as I expect.